### PR TITLE
fix group by over empty result sets

### PR DESCRIPTION
**Pull request by JCOR11599**
fix for: https://github.com/dolthub/dolt/issues/5683

### CODE REVIEW

File: sql/analyzer/aggregations.go [hunk 1]
```diff
@@ -108,11 +108,24 @@ func replaceAggregatesWithGetFieldProjections(_ *sql.Context, scope *Scope, proj
...
-			)
+			// add to plan.GroupBy.SelectedExprs iff expression has an expression.GetField
+			hasGetField := false
+			transform.InspectExpr(e, func(expr sql.Expression) bool {
+				if _, ok := expr.(*expression.GetField); ok {
+					hasGetField = true
+					return true
+				}
+				return false
+			})
```
Comment (max-hoffman) [inline sql/analyzer/aggregations.go:119]:
```suggestion
			// add to plan.GroupBy.SelectedExprs iff expression has an expression.GetField
			hasGetField = transform.InspectExpr(e, func(expr sql.Expression) bool {
				_, ok := expr.(*expression.GetField)
				return ok
			})
```

File: enginetest/queries/query_plans.go [hunk 2]
```diff
@@ -910,6 +910,28 @@ var PlanTests = []QueryPlanTest{
 			"     └─ columns: [a]\n" +
 			"",
 	},
+	{
```
Comment (max-hoffman) [inline enginetest/queries/query_plans.go:913]: can we add some tests 1) with this transform preformed inside subqueries, making sure they stay cached afterwards (`select * from xy join (select count(*) as u, 1 as v from ...) uv on x = u`, and 2)) this transform performed inside subquery expressions, with semi/anti joins still being generated afterwards (` select ... where x in/not in (select y from (select count(*) as u, 1 as v from ...) uv)`, `select ... where x =/!= (...)`, `select * from xy where exists (select y from (select count(*) as u, 1 as v from ...) uv) where x = y)`)

Comment (max-hoffman) [reply enginetest/queries/query_plans.go:913]: and idk if this is valid, but can the projection be a subquery expression? `select count(*), (select * from uv where u = x) from xy group by x`

File: sql/analyzer/resolve_columns.go [hunk 3]
```diff
@@ -1021,7 +1021,7 @@ func pushdownGroupByAliases(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Sc
 		if n.Resolved() || !ok || len(g.GroupByExprs) == 0 {
 			return n, transform.SameTree, nil
 		}
-		if !ok || len(g.GroupByExprs) == 0 {
+		if !ok || len(g.GroupByExprs) == 0 { // TODO: this never happens because of above if???
```
Comment (PavelSafronov) [inline sql/analyzer/resolve_columns.go:1024]: What's the TODO action here?
If the "never happens" situation actually happens, how bad is it? Should we return an error or panic?

Comment (JCOR11599) [reply sql/analyzer/resolve_columns.go:1024]: This was just a note for myself while I was looking through the code.
If we somehow make it to this condition and it evaluates to true, we would just do what we did in the previous block.

I'm pretty sure this block can be safely removed

**Review by max-hoffman — APPROVED** [review]
lgtm

**Pull request merged**